runtime/Cpp/runtime/nuget/pack.cmd

```diff
@@ -0,0 +1,101 @@
+echo off
+rem echo Usage:
+rem echo ------
+rem echo pack vsvers version [pre]      // pack 2019 4.9.1 -beta
+rem echo ------
+setlocal enableextensions enabledelayedexpansion
+
+if "%1"=="" goto usage
+if "%2"=="" goto usage
+set PRE=%3
+set PLATFORM=Win32
+
+rem -version ^^[16.0^^,17.0^^)
+set VS_VERSION=vs%1
+rem  should be set "VSWHERE='%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe  -property installationPath -version ^[16.0^,17.0^)'"
+if %VS_VERSION%==vs2019 (
+  set "VSWHERE='C:\PROGRA~2\"Microsoft Visual Studio"\Installer\vswhere.exe  -latest -property installationPath -version ^[16.0^,17.0^)'"
+) else (
+if %VS_VERSION%==vs2017 (
+  set "VSWHERE='C:\PROGRA~2\"Microsoft Visual Studio"\Installer\vswhere.exe  -latest -property installationPath -version ^[15.0^,16.0^)'"
+)
+)
+for /f " delims=" %%a in (%VSWHERE%) do @set "VSCOMNTOOLS=%%a"
+
+echo ============= %VSCOMNTOOLS% =============
+
+if %VS_VERSION%==vs2019 (
+  set VS_VARSALL=..\..\VC\Auxiliary\Build\vcvarsall.bat
+  set "VS160COMNTOOLS=%VSCOMNTOOLS%\Common7\Tools\"
+) else (
+  if %VS_VERSION%==vs2017 (
+    set VS_VARSALL=..\..\VC\Auxiliary\Build\vcvarsall.bat
+    set "VS150COMNTOOLS=%VSCOMNTOOLS%\Common7\Tools\"
+  ) else (
+    set VS_VARSALL=..\..\VC\vcvarsall.bat
+  )
+)
+
+if not defined VCINSTALLDIR (
+  if %VS_VERSION%==vs2015 (
+    if %PLATFORM%==x64 (
+      call "%VS140COMNTOOLS%%VS_VARSALL%" x86_amd64 8.1
+    ) else (
+      call "%VS140COMNTOOLS%%VS_VARSALL%" x86 8.1
+    )
+  ) else (
+    if %VS_VERSION%==vs2017 (
+      if %PLATFORM%==x64 (
+        call "%VS150COMNTOOLS%%VS_VARSALL%" x86_amd64 8.1
+      ) else (
+        call "%VS150COMNTOOLS%%VS_VARSALL%" x86 8.1
+      )
+    ) else (
+      if %VS_VERSION%==vs2019 (
+        if %PLATFORM%==x64 (
+          call "%VS160COMNTOOLS%%VS_VARSALL%" x86_amd64 8.1
+        ) else (
+          call "%VS160COMNTOOLS%%VS_VARSALL%" x86 8.1
+        )
+      )
+    )
+  )
+)
```

Comment:
Please drop support of VS < 2019 (2017, 2015) and add support of VS2022. It was updated in https://github.com/antlr/antlr4/pull/3725